submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,304 @@
+
+                <div class="user-headings">
+                    <h5><strong>${name}</strong></h5>
+                    <h6>${age} years old</h6>
+                </div>
+
+                <div class="user-description">
+                    <p><strong>User Location:</strong></p>
+                    <p>${birthDate}</p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Location:</strong></p>
+                    <p>${country}</p>
+                    <p>${state}</p>
+                    <p>${city}</p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Contacts:</strong></p>
+                    <p>
+                        <a href="mailto:${email}">${email}</a>
+                    </p>
+                    <p>
+                        <a href="tel:${phone}">${phone}</a>
+                    </p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Gender:</strong></p>
+                    <p class="uppercase">${gender}</p>
+                </div>
+
+                <div class="user-button user-button_close">
+                    Back to Friends
+                </div>
+            </div>
+        </div>
+    `;
+}
+
+// filters
+const sidebarFilters = document.querySelector('.main__sidebar');
+sidebarFilters.addEventListener("submit", function(event) {
+    event.preventDefault();
+
+    renderUsers(getSidebarOptions(), usersList);
+
+    if (window.screen.width <= 992) {
+        toggleMenu();
+    }
+});
+
+const getSidebarOptions = () => {
+    const formData = new FormData(sidebarFilters);
+    const filtersObject = {};
+    formData.forEach((value, key) => {
+        filtersObject[key] = value;
+    });
+
+    return filtersObject;
+}
+
+const filterUsers = (filtersObject, usersArray) => {
+    const pipes = [filterByName, filterByGender, sortBy];
+    const filterName = filtersObject.name;
+    const filterGender = filtersObject.gender;
+    const filterSort = filtersObject.filterSort;
+
+    let users = pipes.reduce((list, func) => {
+        switch (func.name) {
+            case 'filterByName':
+                return func(list, filterName);
+            case 'filterByGender':
+                return func(list, filterGender);
+            case 'sortBy':
+                return func(list, filterSort);
+            default:
+                return usersArray;
+        }
+    }, usersArray);
+
+    usersFilteredList = users;
+
+    return users;
+}
+
+const filterByName = (array, searchValue) => {
+    if (searchValue) {
+        searchValue = searchValue.trim().toLowerCase();
+        return array.filter(user => user.name.first.toLowerCase().includes(searchValue));
+    } else {
+        return array;
+    }
+}
+
+const filterByGender = (array, filterOption) => {
+    if (filterOption !== 'genderAll') {
+        return array.filter(user => user.gender === filterOption);
+    } else {
+        return array;
+    }
+}
+
+const sortBy = (array, sortValue) => {
```

Comment:
You can make this code even more dry. Both sortings are using the same principle, they get value A, value B, compare it and return -1, 0 or 1. It makes sense for both letters and numbers. So essentially you can have only 1 sorting functions. The trick is to feed it the right arguments. Try to think about how you can do it.